### PR TITLE
Kevin/dep updates

### PR DESCRIPTION
**Pull request by knisbet**
Update client-go to release 6 (k8s 1.9)
Also, redo @UnderGreen change from upper to lower case on logrus, and update vendoring.

### CODE REVIEW

File: Gopkg.toml [hunk 1]
```diff
@@ -20,34 +20,34 @@
...
-  name = "gopkg.in/check.v1"
+[[override]]
+  name = "k8s.io/api"
+  branch = "release-1.9"
 
-[[constraint]]
-  branch = "release-1.7"
+[[override]]
   name = "k8s.io/apimachinery"
+  revision = "5134afd2c0c91158afac0d8a28bd2177185a3bcc"
```
Comment (r0mant) [inline Gopkg.toml]: Do we need a particular revision here, or can use `release-1.9` branch?

File: changeset.go [hunk 2]
```diff
@@ -1559,43 +1557,6 @@ func (cs *Changeset) upsertSecret(ctx context.Context, tr *ChangesetResource, da
...
-	if err != nil {
-		return trace.Wrap(err)
-	}
-	versionMinor, err := strconv.Atoi(version.Minor)
-	if err != nil {
-		return trace.Wrap(err)
-	}
-
-	// TODO(knisbet) deprecated. Remove Third Party resources when we're no longer supporting kubernetes < 1.8
-	if versionMajor == 1 && versionMinor < 8 {
```
Comment (r0mant) [inline changeset.go]: Removing all this means that if 4.x will need a fix in rigging, we'll have to branch out from one of older tags. Maybe keep all this for the meanwhile? Technically we still have (and will have, for foreseeable future) 1.7 running in production.

File: Gopkg.toml (hunk 1 above)
Comment (knisbet) [reply Gopkg.toml]: The release-1.9 branch wasn't compatible when I tried it yesterday, and iirc this is what's currently in the glide.lock file.

File: changeset.go (hunk 2 above)
Comment (knisbet) [reply changeset.go]: I think the third party resources was removed from the library... at least I can't seem to find it. Although I haven't tested it, Custom Resources was considered stable in 1.7.

**Comment by a-palchikov** [comment]
I keep getting substantial differences when I `dep ensure -v && dep prune` - which version of deps were these changes from?

**Comment by knisbet** [comment]
it's also possible that the upstream branch has had changes between when I pulled them in and when you did.
```
dep:
 version     : v0.4.1
 build date  : 2018-01-27
 git hash    : 37d9ea0
 go version  : go1.9.3
 go compiler : gc
 platform    : darwin/amd64
```

**Pull request merged**